modules/rabbitmq/src/main/java/org/apache/axis2/transport/rabbitmq/RabbitMQMessageSender.java

```diff
@@ -19,315 +19,264 @@
-        this.targetEPR = targetEPR;
-        this.connectionFactory = factory;
+    public RabbitMQMessageSender(Channel channel, String factoryName, RabbitMQSender.SenderType senderType) {
+        this.channel = channel;
+        this.senderType = senderType;
+        this.factoryName = factoryName;
+    }
 
-        try {
-            rmqChannel = connectionFactory.getRMQChannel();
-        } catch (InterruptedException e) {
-            handleException("Error while getting RPC channel", e);
+    /**
+     * Publish message to the exchange with the routing key. Execute relevant logic based on the sender type.
+     *
+     * @param routingKey         the routing key to publish the message
+     * @param msgContext         the axis2 message context
+     * @param rabbitMQProperties the rabbitmq endpoint parameters
+     * @return {@link Delivery} in RPC messaging style
+     * @throws AxisRabbitMQException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    // TODO: handle x-consistent-hash
+    public Delivery send(String routingKey, MessageContext msgContext, Hashtable<String, String> rabbitMQProperties)
+            throws IOException {
+        Delivery response = null;
+
+        // declaring queue if given
+        String queueName = rabbitMQProperties.get(RabbitMQConstants.QUEUE_NAME);
+        RabbitMQUtils.declareQueue(channel, queueName, rabbitMQProperties);
+
+        // declaring user defined exchange if given
+        String exchangeName = rabbitMQProperties.get(RabbitMQConstants.EXCHANGE_NAME);
+        RabbitMQUtils.declareExchange(channel, exchangeName, rabbitMQProperties);
+
+        AMQP.BasicProperties.Builder builder = buildBasicProperties(msgContext);
+
+        int deliveryMode = NumberUtils.toInt(rabbitMQProperties.get(RabbitMQConstants.QUEUE_DELIVERY_MODE),
+                RabbitMQConstants.DEFAULT_DELIVERY_MODE);
+        builder.deliveryMode(deliveryMode);
+
+        long replyTimeout = NumberUtils.toLong((String) msgContext.getProperty(RabbitMQConstants.RABBITMQ_WAIT_REPLY),
+                RabbitMQConstants.DEFAULT_RABBITMQ_TIMEOUT);
+
+        long confirmTimeout = NumberUtils.toLong((String) msgContext.getProperty(RabbitMQConstants.RABBITMQ_WAIT_CONFIRMS),
+                RabbitMQConstants.DEFAULT_RABBITMQ_TIMEOUT);
+
+        AMQP.BasicProperties basicProperties = builder.build();
+        byte[] messageBody = RabbitMQUtils.getMessageBody(msgContext);
+
+        switch (senderType) {
+            case RPC:
+                response = sendRPC(exchangeName, routingKey, basicProperties, messageBody, replyTimeout);
+                break;
+            case PUBLISHER_CONFIRMS:
+                sendPublisherConfirms(exchangeName, routingKey, basicProperties, messageBody, confirmTimeout);
+                break;
+            case DEFAULT:
+                publishMessage(exchangeName, routingKey, basicProperties, messageBody);
+                break;
```

Comment:
I think we can replace this with the default case in the switch statement itself.
```suggestion
            default:
                publishMessage(exchangeName, routingKey, basicProperties, messageBody);
                break;
```

Reply:
Ack.